backend/tests/helpers/utils.py

```diff
@@ -35,6 +36,28 @@ def start_neo4j_container(neo4j_image: str) -> DockerContainer:
     return container
 
 
+def start_prefect_server_container(
+    request: pytest.FixtureRequest,
+) -> dict[int, int] | None:
+    if not INFRAHUB_USE_TEST_CONTAINERS:
+        return None
+
+    container = (
+        DockerContainer(image="prefecthq/prefect:3.0.11-python3.12")
+        .with_command("prefect server start --host 0.0.0.0 --ui")
+        .with_exposed_ports(PORT_PREFECT)
+    )
+
+    def cleanup() -> None:
+        container.stop()
+
+    container.start()
+    wait_for_logs(container, "Configure Prefect to communicate with the server")
+    request.addfinalizer(cleanup)
```

Comment:
Does the request.addfinalizer work in the same way when this function isn't a fixture? I'm not sure just wondering if the fixture decorator adds anything around this or if it's managed through the request object itself?

Reply:
I guess fixture decorator checks request object state when fixture function scope ends. So whereever we are mutating this object should not matter